lib/src/repository/traits/thermostat_repository.dart

```diff
@@ -30,6 +30,8 @@ class ThermostatRepository {
         FanMode(trait.state.fanMode.reported?.value ?? AvailableFanMode.ON),
         ThermostatMode(
             trait.state.mode.reported?.value ?? AvailableThermostatMode.OFF),
+        AmbientTemperature(
+            trait.state.ambientTemperature.reported?.value ?? 0.0),
```

Comment:
default to null???

Reply:
If this is the case would we want to include the state at all in the trait?

Reply:
If we didn't include the state then we would probably want to make it's argument non nullable and then we would only handle the states that are present.